drivers/gles3/rasterizer_scene_gles3.cpp

```diff
@@ -1121,13 +1121,19 @@ bool RasterizerSceneGLES3::_setup_material(RasterizerStorageGLES3::Material *p_m
 
 	if (state.current_depth_draw != p_material->shader->spatial.depth_draw_mode) {
 		switch (p_material->shader->spatial.depth_draw_mode) {
-			case RasterizerStorageGLES3::Shader::Spatial::DEPTH_DRAW_ALPHA_PREPASS:
+			case RasterizerStorageGLES3::Shader::Spatial::DEPTH_DRAW_ALPHA_PREPASS: {
+				glDepthMask(GL_TRUE);
+				// If some transparent objects write to depth, we need to re-copy depth texture when we need it
+				if (p_alpha_pass && !state.used_depth_prepass) state.prepared_depth_texture = false;
```

Comment:
Please avoid putting condition and statement on the same line, it's not very readable.